core/services/workflows/engine_test.go

```diff
@@ -729,6 +731,7 @@ func TestEngine_WrapsTargets(t *testing.T) {
 	require.NoError(t, err)
 
 	<-hooks.initSuccessful
+	eng.Close()
```

Comment:
This one is still trouble if the test times out while waiting for the channel receive.

Reply:
Thanks for flagging! I do want to close there though, so I added a servicetest.Run above it; this will lead to a redundant Close() call in the happy path case, but should be safe since we StopOnce it.

Reply:
Yes, thinking about it a bit more, I don't actually need to close exactly there as the workflow is never mutated after initialization and we only read from it. So I think there's no race that can happen here and we can safely just `servicetest.Run`